client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -115,18 +119,45 @@ class JetpackChecklist extends PureComponent< Props > {
 					completedTitle={ task.completedTitle }
 					description={ task.description }
 					duration={ task.duration }
+					firstIncomplete={ this.getFirstIncompleteTask( taskIds ) }
 					href={ task.getUrl( this.props.siteSlug, isComplete ) }
+					id={ taskId }
+					key={ taskId }
 					onClick={ this.handleTaskStart( {
 						taskId,
 						tourId: get( task, [ 'tourId' ] ),
 					} ) }
+					onTaskTitleClick={ this.onTaskTitleClick }
 					title={ task.title }
-					key={ taskId }
+					selectedTaskId={ this.state.selectedTaskId }
 				/>
 			);
 		} );
 	}
 
+	onTaskTitleClick = selectedTaskId => {
+		this.setState( { selectedTaskId } );
+		this.props.recordTracksEvent( 'calypso_checklist_task_expand', {
+			step_name: selectedTaskId,
+			product: 'Jetpack',
+		} );
+	};
+
+	getFirstIncompleteTask = taskStatuses => {
+		const taskObject = {};
+
+		Object.values( taskStatuses ).find( function( item, index ) {
+			if ( item.completed === false ) {
+				taskObject.id = Object.keys( taskStatuses )[ index ];
+				taskObject.isCompleted = item.completed;
+			}
+
+			return item.completed === false;
+		} );
+
+		return taskObject;
```

Comment:
The `getFirstIncompleteTask` logic could probably be shortened to something like https://github.com/Automattic/wp-calypso/blob/60c089ce54589232d998a0d2f159c7536db8bc40/client/my-sites/checklist/wpcom-checklist/wpcom-task-list.js#L125-L127

Also, it probably makes sense moving that logic into `components/checklist`.